R/further_analytical_functions.R

```diff
@@ -0,0 +1,308 @@
+#' @return a list containing a data.frame with indices according to
+#' Bocquet-Appel and Masset showing the computed exact value as well
+#' as ranges and an object of class mortaar_life_table with the corrected
+#' values.
+#' \itemize{
+#'   \item \bold{e0}:   Corrected life expectancy.
+#'   \item \bold{1q0}:   Mortality of age group 0--1.
+#'   \item \bold{5q0}:   Mortality of age group 0--5.
+#'}
+#'
+#' @examples
+#' # Calculate representativity indices from real life dataset.
+#' library(magrittr)
+#' mag <- magdalenenberg
+#' mag <- mag %>% replace(mag == "60-x", "60-69")
+#' mag <- mag %>% tidyr::separate(a, c("from", "to")) %>%
+#' transform(from = as.numeric(from), to = as.numeric(to))
+#' mag_prep <- mag %>% prep.life.table(dec = "Dx",
+#' agebeg = "from", ageend = "to", method = "Equal5",
+#' agerange = "included")
+#' mag_result <- mag_prep %>% life.table()
+#' lt.correction(mag_result)
+#'
+#' @export
+#'
+lt.correction <- function(life_table, agecor = TRUE, agecorfac = c(), option_spline = NULL) {
+
+  indx <- lt.indices(life_table)
+
+  # corrected life expectancy at birth after Bocquet-Appel and Masset
+  e0 <- 78.721 * log10(sqrt(1 / indx$juvenile_i$d5_14_d20plus)) - 3.384
+  e0_range <- paste0(round(e0 - 1.503,1), "--", round(e0 + 1.503,1))
+
+  # corrected mortality for age group 1q0 after Bocquet-Appel and Masset
+  q1_0 <- 0.568 * sqrt((log10(200 * indx$juvenile_i$d5_14_d20plus))) - 0.438
+  q1_0_range <- paste0(round(q1_0 - 0.016,3), "--", round(q1_0 + 0.016,3))
+
+  # corrected mortality for age group 5q0 after Bocquet-Appel and Masset
+  q5_0 <- 1.154 * sqrt((log10(200 * indx$juvenile_i$d5_14_d20plus))) - 1.014
+  q5_0_range <- paste0(round(q5_0 - 0.041,3), "--", round(q5_0 + 0.041,3))
+
+  Dx_sum_corrected <- (life_table$Dx %>% sum - life_table$Dx[1]) / (1 - q5_0)
+  Dx5_0_corrected <- q5_0 * Dx_sum_corrected
+
+  if (5 == life_table$a[1]) {
+    life_table$Dx[[1]] <- Dx5_0_corrected
+  } else if ((4 == life_table$a[2]) & (1 == life_table$a[1])) {
+    Dx1_0_corrected <- q1_0 * Dx_sum_corrected
+    life_table$Dx[[1]] <- Dx1_0_corrected
+    life_table$Dx[[2]] <- Dx5_0_corrected - Dx1_0_corrected
+  } else {
+    stop("Life table correction works only with one 5-year-age class or 1- and 4-year classes
+         for the first 5 years. Please take a look at ?life.table to determine how your
+         input data should look like for accomplishing this.")
+  }
+  life_table_prep_corrected <- data.frame(cbind(a = life_table$a, Dx = life_table$Dx))
+  life_table_corr <- life.table(life_table_prep_corrected, agecor = agecor, agecorfac = agecorfac, option_spline = option_spline)
+  row_e0 <- c(round(e0, 1), e0_range)
+  row_q1_0 <- c(round(q1_0, 3), q1_0_range)
+  row_q5_0 <- c(round(q5_0, 3), q5_0_range)
+  e0_q5_0 <- data.frame(rbind(row_e0, row_q1_0, row_q5_0))
+  colnames(e0_q5_0) <- c("value", "range")
+  rownames(e0_q5_0) <- c("e0", "1q0", "5q0")
+  return(list(indices = e0_q5_0, life_table_corr = life_table_corr))
+  }
```

Comment:
Some more comments in the code would be nice.

Reply:
Added comments.